### DISCUSSION TITLE
리액트 쿼리을 SWR로 바꾸는 것에 대한 논의

### DISCUSSION THREAD

**Discussion by chsua** [woowacourse-teams/2023-votogether#489] [자유로운 토론]
처음에는 리액트 쿼리의 다양한 기능을 사용하고자 선택하였으나
러닝 커브 등의 이유로 충분한 활용을 하고 있지 않다고 느낍니다.
더불어 SWR이 리액트쿼리에 비해 크기가 작고 쿼리 키를 삭제하여 데이터를 가지고 오는 것이
더 빠르기 때문에 SWR로 바꾸는 것에 대해 논의를 해보면 좋을 것 같습니다. 

각 라이브러리의 장단점과 변화에 따른 코드 수정 부분을 고려하여 결정해보아요!

**Comment by Gilpop8663** [comment]
참고 자료를 찾을 수록 비슷한 역할을 하는 패키지들이라서 고민이 되네요

현재 프로젝트에서 swr을 도입했을 때 이점이  아래와 같이 느껴졌어요. 그래서 하지 않았을 때보다 하였을 때 얻는 점이 많다고 생각해서 바꿔보는 것이 좋다고 생각이 들었습니다
1. 번들 사이즈가 가벼워 초기 로딩이 빨라진다.
2. swr과 react-query의 차이점을 체감할 수 있다 ( 학습적 측면 )
3. react-query를 swr로 마이그레이션 해보며 패키지를 사용하는 코드를 마이그레이션 하는 경험을 해볼 수 있다.

**Comment by inyeong-kang** [comment]
음 저희가 초반에 기술스택 정할 때, react query를 사용하자! 라고 한 이유들로
1. 무한스크롤(useInfiniteQuery)
2. 캐시 기능(cacheTime 조절 가능) 
이 떠오르네요!

특히 무한스크롤 관점에서 찾아봤는데, swr를 쓰면 페이지네이션하는 코드를 별도로 작성해줘야 하는거 같더라구요..?

더불어 tanstack query는 있지만 swr는 없는 중요한 기능으로,
1. 가비지 컬렉션
2. 네트워크 끊어지면 refetch해주기
가 있다고 하네요.. 

Swr가 가볍지만, 그만큼 아직은 query에 비해 부족한 기능들이 있다는 생각이 듭니다🤔

저희 핵심 기능인 게시글 무한스크롤 기능에, swr를 적용해보고 판단하는 것도 좋겠네요! (마이그레이션 연습은 저희 repo fork 떠서 해보기? 어떠신지..)

참고: https://tech.madup.com/react-query-vs-swr/
[useSWRInfinite](https://velog.io/@jungmiin/Intersection-Observer-API%EB%A1%9C-%EB%AC%B4%ED%95%9C-%EC%8A%A4%ED%81%AC%EB%A1%A4-%EA%B5%AC%ED%98%84%ED%95%98%EA%B8%B0)
